### PR TITLE
Added bl0940 documentation

### PR DESCRIPTION
**Pull request by tobias-**
## Description:

This adds documentation for the new BL0940 integration.

**Related issue (if applicable): fixes esphome/feature-requests#770

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable): esphome/esphome#1904

## Checklist:

  - [X] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [X] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

**Comment by tobias-** [comment]
@oxan this is also ready to review. The check that fails was cancelled for some reason being my control. At least at that time, no Deploy preview builds seemed to succeed.

File: components/sensor/bl0940.rst [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    :image: bl0940.png
+    :keywords: BL0940
+
+.. note::
+
+    This page is incomplete and could some work. If you want to contribute, please read the
+    :doc:`contributing guide </guides/contributing>`. This page is missing:
+
+      - An image for the front page.
+      - Images/screenshots/example configs of this device being used in action.
```
Comment (oxan) [inline components/sensor/bl0940.rst:14]: This seems outdated.

**Review by oxan — COMMENTED** [review]
LGTM otherwise.

File: components/sensor/bl0940.rst (hunk 1 above)
Comment (tobias-) [reply components/sensor/bl0940.rst:14]: sorry about that. Will fix immediately

**Review by oxan — APPROVED** [review]

**Pull request merged**